hack/generator/pkg/functions/original_version_function.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package functions
+
+import (
+	"github.com/dave/dst"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astbuilder"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+)
+
+// OriginalVersionFunction implements a function to return the version for the type on which it is called
+// We put these on the *API* versions of our spec types, giving us a way to obtain the right type of instance when the
+// reconciler is working with ARM.
+//
+// func (spec *SomeSpec) OriginalVersion() string {
+//     return GroupVersion.Version
+// }
+//
+type OriginalVersionFunction struct {
+	idFactory astmodel.IdentifierFactory
+}
+
+// Ensure OriginalVersionFunction properly implements ValueFunction
+var _ astmodel.ValueFunction = &OriginalVersionFunction{}
+
+// NewOriginalVersionFunction creates a new OriginalVersionFunction
+func NewOriginalVersionFunction(idFactory astmodel.IdentifierFactory) *OriginalVersionFunction {
+	return &OriginalVersionFunction{
+		idFactory: idFactory,
+	}
+}
+
+// Name returns the name of this function, which is always OriginalVersion()
+func (o OriginalVersionFunction) Name() string {
```

Comment:
I think since you're returning a `*OriginalVersionFunction` and asserting `var _ astmodel.ValueFunction = &OriginalVersionFunction {}` that the receiver here should also be a ptr (this applies to all the functions here)

Reply:
Fixed.